### PR TITLE
ref: Update All fn Selector "Magic Values" Into Explicit fn Selector Calculation

### LINKED ISSUE #419: [Feature]: Update All fn Selector "Magic Values" Into Explicit fn Selector Calculation
### Example

Turn this:

```rust
 const SINGLE_TRANSFER_FN_SELECTOR: FixedBytes<4> = fixed_bytes!("f23a6e61");
```

Into this:

```rust
const SINGLE_TRANSFER_FN_SELECTOR: [u8; 4] = function_selector!(
    "onERC1155Received",
    Address,
    Address,
    U256,
    U256,
    Bytes
);
```

Includes [these types of magic values](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/token/erc721/extensions/metadata.rs#L60):

```rust
    fn supports_interface(interface_id: FixedBytes<4>) -> bool {
        // NOTE: interface id is calculated using additional selector
        //  [`Erc721Metadata::token_uri`]
        0x_5b5e139f == u32::from_be_bytes(*interface_id)
    }
```

Original comment that prompted this issue's creation: https://github.com/OpenZeppelin/rust-contracts-stylus/pull/417?notification_referrer_id=NT_kwDOAShu7bQxMzQ5NzI1NDA0ODoxOTQyNzA1Mw#discussion_r1856153429

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by 0xNeshi**
Resolves #419

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Changelog

### CODE REVIEW

File: contracts/src/token/erc721/extensions/metadata.rs [hunk 1]
```diff
@@ -21,6 +21,9 @@ sol_storage! {
     }
 }
 
+// TODO: IErc721Metadata should be refactored to have same api as solidity
```
Comment (bidzyyys) [inline contracts/src/token/erc721/extensions/metadata.rs:24]: Remove this comment. We were facing this issue with `v0.1` and decided internally that this is the best way for implementing this extension.

**Review by bidzyyys — COMMENTED** [review]
Great job @0xNeshi. Left small comment.

**Review by bidzyyys — APPROVED** [review]
LGTM!

**Review by qalisander — APPROVED** [review]
Good!

**Pull request merged**